docs/design_docs/metal_hal_driver.md

```diff
@@ -11,6 +11,14 @@ GPU APIs is generally straightforward. This applies to the Metal HAL driver.
 
 ## Overall Design Choices
 
+### Metal Versions
+
+The Metal HAL driver expects Metal 2+. Metal 2 introduces useful features like
+argument buffer, performance shaders, and others, that can improve performance
+and make IREE HAL implementation simpler. Metal 2 was released late 2017.
+Typically Apple software gets updated fairly quickly on various devices, so we
+expect it's already very common now.
```

Comment:
Are there public stats on Metal 2 adoption? "we expect it's already very common" isn't too convincing, coming from an Android/Vulkan/WebGL background :P

Reply:
It's since macOS High Sierra, iOS 11 and tvOS 11 so we can use OS adoption. 

https://gs.statcounter.com/macos-version-market-share/desktop/worldwide (85%+)
https://developer.apple.com/support/app-store/ (95%+)

:)

Reply:
Embedded links in the doc too.